x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -326,6 +326,12 @@ public static String dateTimeToString(long dateTime) {
         return DEFAULT_DATE_TIME_FORMATTER.formatMillis(dateTime);
     }
 
+    public static String nanoTimeToString(long dateTime) {
+        // TODO: It feels like we should have a function for this somewhere in DateFormatter?
+        return DateFormatter.forPattern("strict_date_optional_time_nanos")
+            .format(Instant.ofEpochMilli(dateTime / 1_000_000).plusNanos(dateTime % 1_000_000));
```

Comment:
I wonder if we just want some method that returns an `instant` from a `long` of nanos since epoch. Then we can just use the `forPattern` stuff inline.